### PR TITLE
Fix broken js build after moving HaarCascades to contrib

### PR DESCRIPTION
**Pull request by kaingwade**
The HaarCascades related are not completely cleaned up #25311 after #25198, which [breaks the JavaScript build](https://github.com/opencv/opencv/pull/25198#issuecomment-2031835822). The PR is to fix the issue.

Related PR: opencv/opencv_contrib#3712

```
force_builders=Custom
build_image:Docs=docs-js
build_image:Custom=javascript
buildworker:Custom=linux-1,linux-4,linux-f1
```

### CODE REVIEW

**Comment by kaingwade** [comment]
The last example in the [js_nodejs.markdown](https://github.com/opencv/opencv/blob/5.x/doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown) is deleted too because it uses HaarCascades face detection as a demo. Maybe another new example should be provided?

**Comment by asmorkalov** [comment]
I propose to replace the code with DNN-based FaceDetectorNY: https://docs.opencv.org/4.x/df/d20/classcv_1_1FaceDetectorYN.html.

File: doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown [hunk 1]
```diff
@@ -226,120 +226,3 @@ These C++ sources use standard APIs to access the filesystem and the implementat
 In the browser, this filesystem is emulated in memory while in Node.js there's also the possibility of using the local filesystem directly. This is often preferable since there's no need of copy file's content in memory. This section explains how to do just that, this is, configuring emscripten so files are accessed directly from our local filesystem and relative paths match files relative to the current local directory as expected.
 
 ### The example ###
-
```
Comment (asmorkalov) [inline doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown:229]: The example corresponds to `Dealing with files` chapter.  We cannot just remove the code, it should be replaced by something meaningful.

**Review by asmorkalov — CHANGES_REQUESTED** [review]

File: doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown (hunk 1 above)
Comment (kaingwade) [reply doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown:229]: > The example corresponds to `Dealing with files` chapter. We cannot just remove the code, it should be replaced by something meaningful.

Correct. I tried to replace it with the DNN-based FaceDetectorYN. However, FaceDetectorYN isn't exposed to JS, therefore it has to rewrite the post-processing part in JS. This appears somewhat redundant, @WanliZhong suggests if FaceDetectorYN can be exposed.

**Comment by asmorkalov** [comment]
I added face detector api: https://github.com/opencv/opencv/pull/25430

**Comment by kaingwade** [comment]
I'll replace the example soon.

File: doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown (hunk 1 above)
Comment (kaingwade) [reply doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown:229]: Replaced the Haar based face detection example with FaceDetectorYN.

File: doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown [hunk 2]
```diff
@@ -333,13 +358,15 @@ function installDOM(){
 ### Execute it ###
 
 -   Save the file as `exampleNodeCanvasData.js`.
--   Make sure the files `aarcascade_frontalface_default.xml` and `haarcascade_eye.xml` are present in project's directory. They can be obtained from [OpenCV sources](https://github.com/opencv/opencv/tree/5.x/data/haarcascades).
+-   Make sure the files `face_detection_yunet_2023mar.onnx` are present in project's directory. They can be obtained from [OpenCV Model Zoo](https://github.com/opencv/opencv_zoo/tree/main/models/face_detection_yunet).
```
Comment (asmorkalov) [inline doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown:361]: I propose to add download step as in https://github.com/opencv/opencv/pull/25468 and drop the dependency in the description.

Comment (kaingwade) [reply doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown]: Done.

File: doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown [hunk 3]
```diff
@@ -231,8 +231,57 @@ In the browser, this filesystem is emulated in memory while in Node.js there's a
...
+    download(url, 0);
+  });
+};
+
+if (!existsSync('./face_detection_yunet_2023mar.onnx')) {
+  await createFileFromUrl('./face_detection_yunet_2023mar.onnx', 'https://media.githubusercontent.com/media/opencv/opencv_zoo/main/models/face_detection_yunet/face_detection_yunet_2023mar.onnx')
+}
+
+if (!existsSync('./opencv.js')) {
+  await createFileFromUrl('./opencv.js', 'https://docs.opencv.org/master/opencv.js')
```
Comment (asmorkalov) [inline doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown:278]: please use 5.x branch, but not master. master is shadow copy of 4.x and may be removed soon.

File: doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown [hunk 4]
```diff
@@ -296,6 +345,7 @@ faces.forEach(function(rect) {
 
 const canvas = createCanvas(image.width, image.height);
 cv.imshow(canvas, src);
+console.log('The result is saved.')
```
Comment (asmorkalov) [inline doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown:348]: m.b. `writeFileSync`?

File: doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown [hunk 5]
```diff
@@ -227,52 +227,127 @@ In the browser, this filesystem is emulated in memory while in Node.js there's a
...
+  if(rect.x4>0 && rect.y4>0)
+    cv.circle(src, {x: rect.x4, y: rect.y4}, 2, [255, 0, 255, 255], 2)
+  if(rect.x5>0 && rect.y5>0)
+    cv.circle(src, {x: rect.x5, y: rect.y5}, 2, [0, 255, 255, 255], 2)
+});
+
+const canvas = createCanvas(image.width, image.height);
+cv.imshow(canvas, src);
+console.log('The result is saved.')
+writeFileSync('output3.jpg', canvas.toBuffer('image/jpeg'));
```
Comment (kaingwade) [inline doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown:349]: > m.b. writeFileSync?

File: doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown (hunk 3 above)
Comment (kaingwade) [reply doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown]: Done.

File: doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown (hunk 5 above)
Comment (asmorkalov) [reply doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown:350]: Sorry for typo. I mean print log after write when it's complete.

Comment (kaingwade) [reply doc/js_tutorials/js_setup/js_nodejs/js_nodejs.markdown:349]: Done.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**